### PR TITLE
feat: enable ms defender for created subscriptions

### PR DESCRIPTION
**Pull request by christopherswarren**
Adds security workspace for each subscription
Updates provider scoping to require management sub for log analytics workspace

### CODE REVIEW

File: foundation/shared-services/core/aks.tf [hunk 1]
```diff
@@ -14,6 +14,7 @@ data "azurerm_private_dns_zone" "aks_private_dns_id" {
 }
 
 data "azurerm_log_analytics_workspace" "log_analytics_workspace" {
+  provider            = azurerm.management
```
Comment (delirius325) [inline foundation/shared-services/core/aks.tf:17]: `provider` Doesn't seem to be a valid attribute https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/log_analytics_workspace

File: landing-zones/aks/core/main.tf [hunk 2]
```diff
@@ -32,11 +33,17 @@ data "azurerm_private_dns_zone" "aks_private_dns_id" {
   provider            = azurerm.connectivity
 }
 
-data "azurerm_log_analytics_workspace" "log_analytics_workspace" {
+data "azurerm_log_analytics_workspace" "management" {
+  provider            = azurerm.management
```
Comment (delirius325) [inline landing-zones/aks/core/main.tf:37]: Again, just in case

File: foundation/shared-services/core/aks.tf (hunk 1 above)
Comment (christopherswarren) [reply foundation/shared-services/core/aks.tf:17]: This isn't a resource attribute for the log analytics workspace, but instead a meta-argument. https://www.terraform.io/language/providers/configuration#selecting-alternate-provider-configurations

I think this is valid (and tested)

File: landing-zones/aks/core/main.tf (hunk 2 above)
Comment (christopherswarren) [reply landing-zones/aks/core/main.tf:37]: see above

**Pull request merged**